From d1fc503c4b4b5558d974e0fb7245ed801c734f82 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 1 May 2021 00:37:01 -0700
Subject: [PATCH] dwarf: Don't omit second operand of '?' operator

This is a GNU C extension and is not valid in ISO C.
---
 binutils/ChangeLog |  4 ++++
 binutils/dwarf.c   | 10 +++++++---
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/binutils/ChangeLog b/binutils/ChangeLog
index 8b08bd0db79..43073320aaa 100644
--- a/binutils/ChangeLog
+++ b/binutils/ChangeLog
@@ -1,3 +1,7 @@
+2021-05-01  Michael Forney  <mforney@mforney.org>
+
+	* dwarf.c: Don't omit second operand of '?' operator.
+
 2021-02-06  Nick Clifton  <nickc@redhat.com>
 
 	This is the 2.36.1 release.
diff --git a/binutils/dwarf.c b/binutils/dwarf.c
index 19475e6cec3..f72fc8d8da8 100644
--- a/binutils/dwarf.c
+++ b/binutils/dwarf.c
@@ -3738,9 +3738,13 @@ process_debug_info (struct dwarf_section *           section,
 		  offset_size == 8 ? "64-bit" : "32-bit");
 	  printf (_("   Version:       %d\n"), compunit.cu_version);
 	  if (compunit.cu_version >= 5)
-	    printf (_("   Unit Type:     %s (%x)\n"),
-		    get_DW_UT_name (compunit.cu_unit_type) ?: "???",
-		    compunit.cu_unit_type);
+	    {
+	      const char *name = get_DW_UT_name (compunit.cu_unit_type);
+
+	      printf (_("   Unit Type:     %s (%x)\n"),
+		      name ? name : "???",
+		      compunit.cu_unit_type);
+	    }
 	  printf (_("   Abbrev Offset: 0x%s\n"),
 		  dwarf_vmatoa ("x", compunit.cu_abbrev_offset));
 	  printf (_("   Pointer Size:  %d\n"), compunit.cu_pointer_size);
-- 
2.31.1

